_examples/how-to-use-middy-validation-in-your-serverless-app.md

```diff
@@ -0,0 +1,308 @@
+
+As you can see the endpoint is working absolutely fine. We sent `mani` and `teja` as our `fname` and `lname` respectively and got `Hello, mani-teja` as the response
+
+Now let's remove `lname` from the request object and see what will be the output
+
+![insomnia_post_request_without_correct_schema](/assets/examples/middy-validation-example/request2.png)
+
+Now, you can see the endpoint is working fine but it returned undefined for `lname` as we only sent the `fname` in the request so it returned undefined in the place of `lname`. But in a production app it makes difficult to debug the bugs, so we need to explicitly throw an error when there is a missing info in the request body
+
+## Setting up our middy middleware
+
+We are now ready to use the API we just created. Let's use [Middy validator](https://middy.js.org/packages/validator/) middleware to validate our API.
+
+{%change%} Run the following in the project root.
+
+```bash
+$ npm i --save @middy/core @middy/http-json-body-parser @middy/http-error-handler @middy/validator ajv
+```
+
+Let's understand what the above packages are,
+
+| package                        | explanation                                                                                                                                                                                                                     |
+| ------------------------------ | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| `@middy/core`                  | Core component of the middy framework                                                                                                                                                                                           |
+| `@middy/http-json-body-parser` | This middleware automatically parses HTTP requests with a JSON body and converts the body into an object                                                                                                                        |
+| `@middy/http-error-handler`    | Automatically handles uncaught errors that contain the properties `statusCode` (number) and `message` (string) and creates a proper HTTP response for them (using the message and the status code provided by the error object) |
+| `@middy/validator`             | This middleware automatically validates incoming events and outgoing responses against custom schemas defined with the JSON schema syntax.                                                                                      |
+| `ajv`                          | AJV stands for Another JSON Schema Validator and represents the fastest validator for JSON schemas around                                                                                                                       |
+
```

Comment:
Let's keep the wording for the explanation consistent:
- This middleware parses HTTP requests with a JSON body ...
- This middleware handles uncaught errors that ...
- This middleware validates incoming events and ...

:+1: